cr/api.py

```diff
@@ -88,65 +91,273 @@ def __init__(self, handle: str, token: str):
+        * Checks if ``p`` exists and is a directory.
 
-        * If ``p`` is a directory, check that it appears to contain an AppType
-          project. If Console ``c`` is provided, ask the user to continue.
+        * Check that ``p`` appears to contain an AppType project. If Console
+          ``c`` is provided, ask the user to continue.
 
-        Raises ``FileNotFoundError`` or ``UserCancelError``.
+        * For Django-based projects, check for a correct manage.py, wsgi.py, and
+          settings files.
+
+        Raises FileNotFoundError, NotADirectoryError, or UserCancelError.
         """
         if not p.exists():
             raise FileNotFoundError(p)
-        if p.is_dir():
-            is_project = False
-            if self.app_type in [
-                AppType.CODEREDCMS,
-                AppType.DJANGO,
-                AppType.WAGTAIL,
-            ]:
-                f = "manage.py"
-            elif self.app_type == AppType.WORDPRESS:
-                f = "wp-config.php"
-            elif self.app_type == AppType.HTML:
-                f = "index.html"
-
-            # Check for file.
-            if (p / f).exists():
-                is_project = True
-
-            # Log or display a warning if file is not found.
-            if not is_project:
-                LOGGER.warning(
-                    "`%s` does not appear to contain a %s project.",
-                    p,
-                    self.app_type_name,
-                )
-                if (
-                    c
-                    and "y"
+        if not p.is_dir():
+            raise NotADirectoryError(f"Expected a directory: `{p}`")
+
+        # Check for app_type's most obvious file in project root.
+        if self.app_type in [
+            AppType.CODEREDCMS,
+            AppType.DJANGO,
+            AppType.WAGTAIL,
+        ]:
+            project_file = p / "manage.py"
+        elif self.app_type == AppType.WORDPRESS:
+            project_file = p / "wp-config.php"
+        elif self.app_type == AppType.HTML:
+            project_file = p / "index.html"
+        else:
+            raise Exception(f"Invalid AppType `{self.app_type}`.")
+        project_file_relative = project_file.relative_to(p)
+        if not project_file.is_file():
+            LOGGER.warning(
+                "%s project missing file `%s`.",
+                self.app_type_name,
+                project_file,
+            )
+            if (
+                c
+                and "y"
+                != c.input(
+                    f"Your {self.app_type_name} project is missing a "
+                    f"`{project_file_relative}` file. "
+                    "Without this your app will not deploy correctly. "
+                    r"Continue anyways? [prompt.choices]\[y/N][/] ",
+                ).lower()
+            ):
+                raise UserCancelError()
+
+        # The following checks are only for Django-based projects.
+        if self.app_type not in [
```

Comment:
Django app types is used in a few places, we should move that to some kind of global for ease of use.

Reply:
I refactored it a bit so the apptype checks are in one place.